### ISSUE TITLE
have `--debug` output ignored files as well

### ISSUE THREAD

**Issue by ctb** [getpelican/pelican#2915]
Hi all,

thank you SO VERY MUCH for pelican! I use it for my blog, http://ivory.idyll.org/blog/, and love it :). Hence this issue!

tl;dr it would be nice if `--debug` printed out a message about files it was ignoring as well.

I decided to try upgrading my pelican install using conda, and noticed that none of my recent blog posts were showing up. After a fair amount of poking around, I realized that it was not picking up any of the .md files, because I'd forgotten to install `Markdown`. Great, all my fault, easy to fix!

However, it is a bit odd to me that when I used `--debug` it didn't tell me anything about the files it was ignoring. That would have helped me get to my solution faster. So I thought I'd drop a feature request in the pelican repo about it.

thanks again,
--titus

**Comment by ctb** [comment]
(not that it matters, but here's my blog source - https://github.com/ctb/titus-blog)

**Comment by avaris** [comment]
That would create too much noise when content and static are [in same place](https://docs.getpelican.com/en/latest/content.html#mixed-content-in-the-same-directory), because content reader will pick up what it *can* process, leave rest (a.k.a. ignored) for other generators, like Static generator. We have a warning that lists active readers but only when you it detects nothing can be processed with the active readers (like below). As a compromise we can print out this in debug mode always, which should be enough of a hint to nudge towards a solution (?).

```
WARNING: No valid files found in content for the active readers:
  | BaseReader (static)
  | HTMLReader (htm, html)
  | RstReader (rst)
```

**Comment by ctb** [comment]
On Mon, Aug 30, 2021 at 01:05:10PM -0700, Deniz Turgut wrote:
> That would create too much noise when content and static are [in same place](https://docs.getpelican.com/en/latest/content.html#mixed-content-in-the-same-directory), because content reader will pick up what it *can* process, leave rest (a.k.a. ignored) for other generators, like Static generator. We have a warning that lists active readers but only when you it detects nothing can be processed with the active readers (like below). As a compromise we can print out this in debug mode always, which should be enough of a hint to nudge towards a solution (?).
> 
> ```
> WARNING: No valid files found in content for the active readers:
>   | BaseReader (static)
>   | HTMLReader (htm, html)
>   | RstReader (rst)
> ```

oh, that sounds good! thanks!

is there a way to track which files are completely left on the table?

**Comment by avaris** [comment]
`pelican` internally [tracks it](https://github.com/getpelican/pelican/blob/332be6e5c8005f699a708568939d31dd85e9cffb/pelican/generators.py#L177-L183) but no user facing way to get that. But, a plugin could work that out by comparing the processed files with the ones in content folder and report it.

**Comment by ctb** [comment]
ok. from a purely UX perspective, it seems like having something that says "these files were not processed by any plugin" would be useful. thanks for all the discussion!
